sdk/src/types/block/output/feature/metadata.rs

```diff
@@ -74,6 +74,333 @@ impl core::fmt::Debug for MetadataFeature {
     }
 }
 
+#[cfg(feature = "serde")]
+pub(crate) mod irc_27 {
+    use alloc::collections::{BTreeMap, BTreeSet};
+
+    use getset::Getters;
+    use serde::{Deserialize, Serialize};
+    use url::Url;
+
+    use super::*;
+    use crate::types::block::address::Bech32Address;
+
+    /// The IRC27 NFT standard schema.
+    #[derive(Clone, Debug, Serialize, Deserialize, Getters, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    #[serde(tag = "standard", rename = "IRC27")]
+    #[getset(get = "pub")]
+    pub struct Irc27Metadata {
+        version: String,
+        /// The media type (MIME) of the asset.
+        ///
+        /// ## Examples
+        /// - Image files: `image/jpeg`, `image/png`, `image/gif`, etc.
+        /// - Video files: `video/x-msvideo` (avi), `video/mp4`, `video/mpeg`, etc.
+        /// - Audio files: `audio/mpeg`, `audio/wav`, etc.
+        /// - 3D Assets: `model/obj`, `model/u3d`, etc.
+        /// - Documents: `application/pdf`, `text/plain`, etc.
+        #[serde(rename = "type")]
+        media_type: String,
+        /// URL pointing to the NFT file location.
+        uri: Url,
+        /// The human-readable name of the native token.
+        name: String,
+        /// The human-readable collection name of the native token.
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        collection_name: Option<String>,
+        /// Royalty payment addresses mapped to the payout percentage.
+        #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
+        royalties: BTreeMap<Bech32Address, f64>,
+        /// The human-readable name of the native token creator.
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        issuer_name: Option<String>,
+        /// The human-readable description of the token.
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        description: Option<String>,
+        /// Additional attributes which follow [OpenSea Metadata standards](https://docs.opensea.io/docs/metadata-standards).
+        #[serde(default, skip_serializing_if = "BTreeSet::is_empty")]
+        attributes: BTreeSet<Attribute>,
+    }
+
+    impl Irc27Metadata {
+        pub fn new(media_type: impl Into<String>, uri: Url, name: impl Into<String>) -> Self {
+            Self {
+                version: "v1.0".to_owned(),
```

Comment:
Also add `with_version()` so users can have a higher version without requiring a library update? (if no the other fields don't require a breaking change then)

Reply:
But if the version changes, wouldn't the entire struct be different?

Reply:
Don't know, but maybe there can be a minor version change which keeps the structure the same, but adds some special handling with the data in some of the fields? Just sounds better to not have it hardcoded to this specific version

Reply:
Okay, but in that case why would a user change the version?